### PR TITLE
refactor(python,rust): More cleanup around `arange`

### PR DESCRIPTION
**Pull request by stinodego**
Changes:
* Add `dtype` argument for `int_range`. Right now it's just a cast after the fact - but I will try to implement this later such that the Series gets materialized in the right data type.
* Use `int_range` instead of `arange` internally.
* Update `arange` docs with a proper deprecation message and "See Also" section.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report_python.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ body:
 
         pl.DataFrame({
             "foo": [None, 1, 2],
-            "bar": np.arange(3)
+            "bar": np.int_range(3)
```
Comment (MarcoGorelli) [inline .github/ISSUE_TEMPLATE/bug_report_python.yml:44]: this doesn't exist 😉

**Review by MarcoGorelli — COMMENTED** [review]
minor comment, rest looks good

File: .github/ISSUE_TEMPLATE/bug_report_python.yml (hunk 1 above)
Comment (stinodego) [reply .github/ISSUE_TEMPLATE/bug_report_python.yml:44]: Woops 😄

Comment (stinodego) [reply .github/ISSUE_TEMPLATE/bug_report_python.yml]: Fixed :)

**Review by MarcoGorelli — APPROVED** [review]
Nice one!

**Pull request merged**